src/EnergyPlus/SurfaceGeometry.cc

```diff
@@ -6350,6 +6351,47 @@ namespace SurfaceGeometry {
 		// Formats
 		static gio::Fmt Format_725( "('Surface Heat Transfer Algorithm, ',A,',',A,',',A,',',A)" );
 
+		// For evaporative cooling surfaces 
+		cCurrentModuleObject = "SurfaceProperty:HeatBalanceSourceTerm";
+		int CountAddHeatSourceSurf = GetNumObjectsFound( cCurrentModuleObject );
+		bool OutsideAddHeatSourceTerm;
+		int AdditionalHeatSourceTermSch;
+		for ( Item = 1; Item <= CountAddHeatSourceSurf; ++Item ) {
+			GetObjectItem( cCurrentModuleObject, Item, cAlphaArgs, NumAlphas, rNumericArgs, NumNumbers, IOStatus, lNumericFieldBlanks, lAlphaFieldBlanks, cAlphaFieldNames, cNumericFieldNames );
+			ErrorsFoundSingleSurf = false;
+			Found = FindItemInList( cAlphaArgs( 1 ), Surface, TotSurfaces );
+			OutsideAddHeatSourceTerm = true;
+
+			if ( Found == 0 ) {
+				ShowSevereError( cCurrentModuleObject + "=\"" + cAlphaArgs( 1 ) + "\", did not find matching surface." );
+				ErrorsFoundSingleSurf = true;
+			}
+
+			if ( ! lAlphaFieldBlanks( 2 ) && SameString( cAlphaArgs( 2 ), "Inside" ) ) {
+				OutsideAddHeatSourceTerm = false;
+			}
+
+			if ( ! lAlphaFieldBlanks( 3 ) ) {
+				AdditionalHeatSourceTermSch = EnergyPlus::ScheduleManager::GetScheduleIndex( cAlphaArgs( 3 ) );
+				if ( AdditionalHeatSourceTermSch == 0 ) {
+					ShowSevereError( cCurrentModuleObject + "=\"" + cAlphaArgs( 1 ) + "\", cannot find the matching Schedule: " + cAlphaFieldNames( 3 ) + "=\"" + cAlphaArgs( 3 ) );
+					ErrorsFoundSingleSurf = true;
+				}
+			} else {
+				ShowSevereError( cCurrentModuleObject + "=\"" + Surface( Found ).Name + "\", " + cAlphaFieldNames( 3 ) + " is required." );
+				ErrorsFoundSingleSurf = true;
+			}			
+
+			if ( ! ErrorsFoundSingleSurf ) {
+				Surface( Found ).HasAdditionalHeatSourceTerm = true;
+				Surface( Found ).AdditionalHeatSourceTermSch = AdditionalHeatSourceTermSch;
+				Surface( Found ).OutsideAddHeatSourceTerm = OutsideAddHeatSourceTerm;
+			} else {
+				ErrorsFound = true;
+			}
+		} // End_For evaporative cooling surfaces
```

Comment:
This looks good except you should move the OutsideAddHeatSourceTerm and AdditionalHeatSourceTermSch variable declarations to inside the loop since they aren't used outside of it or from one loop iteration to the next.

Reply:
We modified this according to the updated variables.